modules/end-user-guide/pages/adding-a-vscode-extension.adoc

```diff
@@ -71,52 +85,49 @@ For example, override some container settings, like the memory limit for the Red
       sidecar:  
         memoryLimit: 2Gi
 ----
-. {prod-short} will automatically install the Red Hat Visual Studio Code Java extension in the {prod-short} workspace and increase the memory of the workspace by two gigabytes.
 
-=== Override default preferences
+=== Overriding default preferences
 
 Override the default preferences of the Visual Studio Code extension for the workspace.
 
-.Procedure
+.Prerequisites
+. You have the `.che/che-theia-plugins.yaml` file in the root of the GitHub repository.
 
-. Add `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the extension.
-
-. Specify the preferences in the `Preferences` section.
+.Procedure
+. Add an `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the extension. 
+. Specify the preferences in the `Preferences` section. In the following example, {prod-short} automatically installs Red Hat Visual Studio Code Java extension in the {prod-short} workspace and sets the `java.server.launchMode` preference to `LightWeight`.
 +
+*EXAMPLE*
 [source,yaml,subs="+quotes"]
 ----
   - id: redhat/java  
     override:  
       preferences:
         java.server.launchMode: LightWeight
 ----
-. {prod-short} will automatically install Red Hat Visual Studio Code Java extension in the {prod-short} workspace and sets the `java.server.launchMode` preference to `LightWeight`.
 
-This method allows to override the default preferences or memory limit of the Visual Studio Code extension for the workspace.
-
-Note: It is also possible to define the preferences in the `.vscode/settings.json` file.
+*NOTE* You can also define the preferences in the `.vscode/settings.json` file, either by changing the preferences in the UI of your IDE or by manually adding them to the `.vscode/settings.json` file:
+[source,json,subs="+quotes"]
+----
+{
+"my.preferences": "my-value"
+}
+----
 
 [id="visual-studio-code-extensions-in-devfile"]
-== Defines Visual Studio Code extensions in the devfile
-
-The devfile v2 specification is not allowing to define the Visual Studio Code extensions in the devfile. But it is possible to define some attributes in the devfile.
+== Defining Visual Studio Code extension attributes in the devfile
 
-This definition is helpful when it is not possible to add extra files in the repository but is not the recommended way.
+In case it's not possible to add extra files in the GitHub repository, you can define some of the plug-in attributes by inlining them in the devfile. You can use this procedure with both `.vscode/extensions.json` and `.che/che-theia-plugins.yaml` file contents.
```

Comment:
In case it's not possible to add extra files in the GitHub repository, you can define some of the plug-in or extension attributes by inlining them in the devfile. You can use this procedure with both `.vscode/extensions.json` and `.che/che-theia-plugins.yaml` file contents.